.github/workflows/chekc-fork-branch.yaml

```diff
@@ -0,0 +1,22 @@
+name: Check Fork Branch
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Echo Branch Configuration
+        id: branch_configuration
+        run: |
+          echo "BRANCH_NAME: ${{github.head_ref}}"
+          echo "REPO: ${{github.event.pull_request.head.repo.full_name}}"
+          echo "BASE_REF: ${{github.event.pull_request.base.ref}}"
+      - name: Check fork branch
+        if: github.event.pull_request.head.repo.full_name != github.repository && github.head_ref == 'main'
```

Comment:
Do we need a predicate here to this only gets applied to the kroxylicious repo itself?  I don't think we want to impede, say me creating a PR from my main against your repo.

I am thinking about something like:

        if: ${{ github.repository == 'kroxylicious/kroxylicious' }}

Reply:
> Do we need a predicate here to this only gets applied to the kroxylicious repo itself? I don't think we want to impede, say me creating a PR from my main against your repo.
Why not? If we are trying everybody to work using non-main branches in the forks, I think we should do the same if we want to create a PR to any other forked repo (from yours to mine)

I don't mind to add:
```
github.event.pull_request.head.repo.full_name != 'kroxylicious/kroxylicious'
```
but as mentioned this is only valid from fork to original repo.